src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -13,120 +13,125 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-#ifdef __GNUC__
-# pragma GCC diagnostic ignored "-Wformat-truncation"
-#endif
-
+#include <cstdarg>
+#include <cstring>
 #include <dlfcn.h>
 #include <execinfo.h>
-#include <string.h>
-#include <cstdarg>
 #include <sys/wait.h>
 #include <sys/mman.h>
 #include <sys/prctl.h>
 
-#include "sk_daemon.h"
 #include "core/common/config_reader.h"
 #include "core/common/message.h"
 #include "core/edge/user/shim.h"
+#include "sk_daemon.h"
 
-xclDeviceHandle initXRTHandle(unsigned deviceIndex)
-{
-  return xclOpen(deviceIndex, NULL, XCL_QUIET);
-}
+using severity_level = xrt_core::message::severity_level;
 
-#define STACKTRACE_DEPTH	(25)
-static void stacktrace_logger(const int sig)
+xclDeviceHandle initXRTHandle(unsigned deviceIndex)
 {
-  const int stack_depth = STACKTRACE_DEPTH;
-
-  syslog(LOG_ERR, "%s - got %d\n", __func__, sig);
-  if (sig == SIGCHLD)
-    return;
-  void *array[stack_depth];
-  int nSize = backtrace(array, stack_depth);
-  char **symbols = backtrace_symbols(array, nSize);
-  if (symbols) {
-    for (int i = 0; i < nSize; i++)
-      syslog(LOG_ERR, "%s\n", symbols[i]);
-    free(symbols);
-  }
+  return xclOpen(deviceIndex, nullptr, XCL_QUIET);
 }
 
-xrt::skd* skd_inst = nullptr;
+std::unique_ptr<xrt::skd> skd_inst;
 
 /* Define a signal handler for the child to handle signals */
 static void sigLog(const int sig)
 {
-  if(sig != SIGTERM) {
-    stacktrace_logger(sig);
+
+  switch(sig) {
+    case SIGTERM:
+      {
+	const auto termMsg = "Terminating PS kernel";
+	xrt_core::message::send(severity_level::info, "SKD", termMsg);
+	if(skd_inst)
+	    skd_inst->set_signal(sig);
+      }
+      break;
+    case SIGINT:
+      {
+	const auto intMsg = "Process interrupted";
+	xrt_core::message::send(severity_level::info, "SKD", intMsg);
+	if(skd_inst)
+	    skd_inst->set_signal(sig);
+      }
+      break;
+    default:
+      {
+      std::stringstream st;
+      st << boost::stacktrace::stacktrace();
+      xrt_core::message::send(severity_level::error, "SKD", st.str() );
+	signal (sig, SIG_DFL);
+	kill(getpid(),sig);
```

Comment:
I suspect you will get into a infinite loop here by reposting the signal you just handled

Reply:
This will change signal handler back to default handler, and cause the process to exit